src/rules/no-this-assign-in-render.ts

```diff
@@ -84,21 +92,53 @@ const rule: Rule.RuleModule = {
       inRender = false;
     }
 
+    /**
+     * Walk left side assignment members and return first non-member
+     *
+     * @param {ESTree.MemberExpression} member Member entered
+     * @return {ESTree.Node}
+     */
+    function walkMembers(member: ESTree.MemberExpression): ESTree.Node {
+      if (member.object.type === 'MemberExpression') {
+        return walkMembers(member.object);
+      } else {
+        return member.object;
+      }
+    }
+
     /**
      * Left side of an assignment expr found
      *
      * @param {Rule.Node} node Node entered
      * @return {void}
      */
     function assignmentFound(node: Rule.Node): void {
-      if (!inRender) {
+      if (!inRender ||
+        !propertyMap ||
+        node.type !== 'MemberExpression') {
         return;
       }
 
-      context.report({
-        node: node.parent,
-        messageId: 'noThis'
-      });
+      const nonMember = walkMembers(node) as Rule.Node;
+      if (nonMember.type === 'ThisExpression') {
+        const parent = nonMember.parent as ESTree.MemberExpression;
+        console.log(parent);
+
+        let propertyName = '';
+        if (parent.property.type === 'Identifier' && !parent.computed) {
+          propertyName = parent.property.name;
+        } else if (parent.property.type === 'Literal') {
+          propertyName = String(parent.property.value);
+        }
+
+        if (propertyMap.has(propertyName) ||
+          parent.property.type === 'Identifier' && parent.computed) {
```

Comment:
There doesn't seem to be a great way to get the computed value (i.e. `x` in `this[x]`) to compare against the properties, so we fail on this as well.